packages/editor/src/components/post-preview-button/index.js

```diff
@@ -159,9 +159,14 @@ export class PostPreviewButton extends Component {
 		// https://html.spec.whatwg.org/multipage/interaction.html#dom-window-focus
 		this.previewWindow.focus();
 
-		// If we don't need to autosave the post before previewing, then we simply
-		// load the Preview URL in the Preview tab.
-		if ( ! this.props.isAutosaveable ) {
+		if (
+			// If we don't need to autosave the post before previewing, then we simply
+			// load the Preview URL in the Preview tab.
+			! this.props.isAutosaveable ||
+			// Also, save the post only if user has the post lock.
+			// Do not save or overwrite the post, if the post is already locked.
+			this.props.isPostLocked
```

Comment:
Note: @joemcgill suggested to expose this boolean as a component prop to `PostPreviewButton` but in my opinion, we should never perform any create/update changes to a post which is already locked by someone else. So this should be checked in all the cases. So there's no point exposing it as a prop.